### ISSUE TITLE
[dup] recv(ByteBuffer) fails if message length is bigger than BB len

### ISSUE THREAD

**Issue by davipt** [zeromq/jzmq#257]
Please close. Right issue is at the v3 project: https://github.com/zeromq/jzmq3-x/issues/1
1. create a client that receives messages via socket.recvByteBuffer() or socket.recvZeroCopy and pass a ByteBuffer of capacity 'x'
2. create a server that sends messages with length bigger than 'y'

Then:
- ØMQ spec mentions that when a message is bigger than the expected size, the remaining bytes will be dropped
- the current code seems to implement this correctly (I'll check some stuff and may provide an additional patch, also including preserving and respecting the proper position and limit values)

What happens:
- the JNI code calls the ByteBuffer.position() with the received value from the zmq_send(), which may be bigger than the length value, and hence will throw an exception because position will be bigger than capacity (and limit)

What should happen:
- as the bytes are dropped after the available space is reached, that value is the one that shall be used to re-set the position. 
- in other words, a recvBB or recvNoCopy shall return the ByteBuffer filled up with as much bytes as space permits, but not throw an exception or, fortunately not happening, not crashing
